crmsh/ui_sbd.py

```diff
@@ -0,0 +1,554 @@
+            timeout_dict = self._adjust_timeout_dict(timeout_dict)
+            # merge runtime timeout dict into parameter timeout dict without overwriting
+            timeout_dict = {**self.device_meta_dict_runtime, **timeout_dict}
+
+        watchdog_timeout = timeout_dict.get("watchdog")
+        if watchdog_timeout != self.watchdog_timeout_from_config:
+            update_dict["SBD_WATCHDOG_TIMEOUT"] = str(watchdog_timeout)
+
+        sbd_manager = sbd.SBDManager(
+            device_list_to_init=self.device_list_from_config,
+            timeout_dict=timeout_dict,
+            update_dict=update_dict
+        )
+        sbd_manager.init_and_deploy_sbd()
+        
+    def _configure_diskless(self, parameter_dict: dict):
+        '''
+        Configure diskless SBD based on input parameters and runtime config
+        '''
+        update_dict = {}
+        watchdog_timeout = parameter_dict.get("watchdog")
+        if watchdog_timeout and watchdog_timeout != self.watchdog_timeout_from_config:
+            update_dict["SBD_WATCHDOG_TIMEOUT"] = str(watchdog_timeout)
+        watchdog_device = parameter_dict.get("watchdog-device")
+        if watchdog_device != self.watchdog_device_from_config:
+            update_dict["SBD_WATCHDOG_DEV"] = watchdog_device
+        if not update_dict:
+            logger.info("No change in SBD configuration")
+            return
+
+        sbd_manager = sbd.SBDManager(
+            update_dict=update_dict,
+            diskless_sbd=True
+        )
+        sbd_manager.init_and_deploy_sbd()
+
+    def _device_add(self, devices_to_add: typing.List[str]):
+        '''
+        Implement sbd device add command, add devices to sbd configuration
+        '''
+        all_device_list = self.device_list_from_config + devices_to_add
+        sbd.SBDUtils.verify_sbd_device(all_device_list)
+
+        devices_to_init, _ = sbd.SBDUtils.handle_input_sbd_devices(
+            devices_to_add,
+            self.device_list_from_config
+        )
+
+        logger.info("Append devices: %s", ';'.join(devices_to_add))
+        update_dict = {"SBD_DEVICE": ";".join(all_device_list)}
+        sbd_manager = sbd.SBDManager(
+            device_list_to_init=devices_to_init,
+            update_dict=update_dict,
+            timeout_dict=self.device_meta_dict_runtime
+        )
+        sbd_manager.init_and_deploy_sbd()
+
+    def _device_remove(self, devices_to_remove: typing.List[str]):
+        '''
+        Implement sbd device remove command, remove devices from sbd configuration
+        '''
+        for dev in devices_to_remove:
+            if dev not in self.device_list_from_config:
+                raise self.SyntaxError(f"Device {dev} is not in config")
+        # To keep the order of devices during removal
+        left_device_list = [dev for dev in self.device_list_from_config if dev not in devices_to_remove]
+        if len(left_device_list) == 0:
+            raise self.SyntaxError("Not allowed to remove all devices")
```

Comment:
```suggestion
            raise self.SyntaxError("Not allowed to remove all devices. Run `crm cluster init sbd -S` to bootstrap the diskless-sbd")
```

Intentionally, not give "-F" directly here, to have user think this twice. We can debate this.

Reply:
This is just the interface to remove device, not the interface to configure diskless SBD

Reply:
Or, how about,
"Not allowed to remove all devices. If you do want the diskless-sbd, please run `crm cluster init sbd -S`"

Reply:
"Not allowed to remove all devices" is good enough, we are not clear on what exactly user want, this is my AI complete me:
```
Not allowed to remove all devices. Please use 'crm sbd purge' to remove all devices
```
I think the above case might happen, user just want to purge sbd from cluster